File: TransCoder_tokenized_test_set_functions/python/NTH_PALINDROME_K_DIGITS.py
def nthPalindrome ( n, k ) :
    if ( k & 1 ) :
        temp = k //2
    else:
        temp=k//2-1
    palindrome=10**temp
    palindrome=palindrome+n-1
    print(palindrome,end="")
    if(k&1):
        palindrome=palindrome//10
    while(palindrome):
        print(palindrome%10,end="")
        palindrome=palindrome//10
    

File: TransCoder_tokenized_test_set_functions/java/NTH_PALINDROME_K_DIGITS.java
static void nthPalindrome ( int n, int k ) {
  int temp = ( k & 1 ) != 0 ? ( k / 2 ) : ( k / 2 - 1 );
  int palindrome = ( int ) Math . pow ( 10, temp );
  palindrome += n - 1;
  System . out . print ( palindrome );
  if ( ( k & 1 ) > 0 ) palindrome /= 10;
  while ( palindrome > 0 ) {
    System . out . print ( palindrome % 10 );
    palindrome /= 10;
  }
  System . out . println ( "" );
}